### PR TITLE
Adds metrics-server chart

### PR DESCRIPTION
**Pull request by olemarkus**
**What this PR does / why we need it**:

This PR adds a new chart to the incubator repository for the Kubernetes metrics-server.
https://github.com/kubernetes-incubator/metrics-server

**Special notes for your reviewer**:
I have not come across a chart that adds a new API service so unsure how to do something like that. Right now, this chart just assumes that the API service does not exist.

### CODE REVIEW

**Review by unguiculus — CHANGES_REQUESTED** [review]
Thanks for the PR. Please compare with what you get with `helm create` using Helm 2.8+ and update accordingly. Also, please follow our RBAC best practices.

https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

Is the namespace kube-system required? Otherwise, remove it and let the user decide.

**Comment by kennethaasan** [comment]
Thanks for the feedback @unguiculus. I tried to compare with `helm create` and make changes accordingly. Also applied the RBAC best practices.

Regarding namespace kube-system, it looks like it's required here: https://github.com/kubernetes-incubator/metrics-server/pull/4

File: incubator/metrics-server/Chart.yaml
Comment (unguiculus) [inline incubator/metrics-server/Chart.yaml]: Add maintainer

File: incubator/metrics-server/templates/metrics-server-serviceaccount.yaml
Comment (unguiculus) [inline incubator/metrics-server/templates/metrics-server-serviceaccount.yaml]: `{{- if .Values.serviceAccount.create -}}`

File: incubator/metrics-server/values.yaml
Comment (unguiculus) [inline incubator/metrics-server/values.yaml]: IfNotPresent

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/metrics-server/Chart.yaml
Comment (olemarkus) [reply incubator/metrics-server/Chart.yaml]: Should I just add myself like below? I removed it from the initial commit as it didn't pass the tests.

File: incubator/metrics-server/Chart.yaml
Comment (unguiculus) [inline incubator/metrics-server/Chart.yaml]: You need to specify your Github username

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by DirectXMan12** [comment]
kube-system isn't required for metrics-server itself

**Comment by DirectXMan12** [comment]
metrics-server should also be granted the `system:auth-delegator` clusterrole, and the `extension-apiserver-authentication-reader` role, specifically in the `kube-system` namespace.

**Comment by olemarkus** [comment]
@unguiculus do the changes done in response to your review look okay?

**Comment by unguiculus** [comment]
@DirectXMan12 thanks for reviewing. Could you have another look, please?

**Comment by olemarkus** [comment]
The failure there looks a bit tricky. A plain cluster typically will not have that clusterrole role, but e2e does?

**Comment by olemarkus** [comment]
Should I do anything about the failed test? I see some charts have some sort of default values CI overrides, but i couldn't find any documentation on how this works. If so it would be nice to know which resources the CI server already have.

**Comment by unguiculus** [comment]
The test cluster is on GKE.

Here's the docs you were looking for:
https://github.com/kubernetes/charts/blob/master/test/README.md#providing-custom-test-values

**Comment by olemarkus** [comment]
Thanks. So as GKE comes with metrics-server, I assume everything put into kube-system must be configurable and disabled for the tests. Should it be enabled or disabled by default?

**Comment by unguiculus** [comment]
I guess it makes sense to enable it by default.

**Comment by olemarkus** [comment]
CI passing now. @unguiculus / @DirectXMan12: can you have another look?

File: incubator/metrics-server/templates/cluster-role.yaml
Comment (DirectXMan12) [inline incubator/metrics-server/templates/cluster-role.yaml]: you don't need this unless you're trying to run pod-nanny

**Review by DirectXMan12 — CHANGES_REQUESTED** [review]
nit about permissions -- update deployments is only needed for pod-nanny, IIRC

**Comment by olemarkus** [comment]
Removed the mentioned rule and the server still works.

**Comment by unguiculus** [comment]
Your branch is outdated. Please merge in the latest master to get the latest CI updates. This should help.

File: incubator/metrics-server/templates/NOTES.txt
Comment (unguiculus) [inline incubator/metrics-server/templates/NOTES.txt]: You may want to print a warning if `apiService.create` is `false`.

File: incubator/metrics-server/README.md
Comment (unguiculus) [inline incubator/metrics-server/README.md]: IfNotPresent

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by olemarkus** [comment]
Thanks for the feedback. Have tried to update accordingly. To be honest, this is starting to reach the quality of many of the stable charts now. Would be lovely to merge this soon so more people can test and put this to use.

**Comment by unguiculus** [comment]
I'd suggest you move it to stable before we merge it.

**Review by unguiculus — APPROVED** [review]

**Comment by olemarkus** [comment]
@unguiculus: Cool. Moved!

**Comment by mf-lit** [comment]
@olemarkus I notice the metrics-server deployment container spec doesn't have any provision for setting resources. Before I attempt a PR, I just wanted to make sure I didn't miss something?

**Comment by olemarkus** [comment]
You didn't miss anything. A PR would for this would be good. Thanks

**Comment by mf-lit** [comment]
https://github.com/kubernetes/charts/pull/6488

**Comment by sebastien-prudhomme** [comment]
@olemarkus are you sure on the file cluster-role.yaml?

If i compare to deploy files in the metrics-server github project (https://github.com/kubernetes-incubator/metrics-server/blob/master/deploy/1.8%2B/resource-reader.yaml), some rights are missing:

```diff
@@ -14,9 +14,18 @@ rules:
     resources:
       - pods
       - nodes
+      - nodes/stats
       - namespaces
     verbs:
       - get
       - list
       - watch
+  - apiGroups:
+    - "extensions"
+    resources:
+      - deployments
+    verbs:
+      - get
+      - list
+      - watch
 {{- end -}}
```

Doest the upstream YAML file give too many rights?

**Comment by olemarkus** [comment]
Yes. The upstream one has permissions that is noecessary only with pod-nannny. Was discussed earlier in this PR.

**Comment by sebastien-prudhomme** [comment]
@olemarkus Not this one?

https://github.com/kubernetes-incubator/metrics-server/pull/59

**Comment by olemarkus** [comment]
In K8s 1.11, some RBAC stuff shifted and from then on you do need the `nodes/stats` resource permission. So that one should get merged in as well.

**Pull request merged**
